lib/improver/blending/weighted_blend.py

```diff
@@ -92,7 +92,7 @@ def conform_metadata(
             Cube containing the adjusted metadata.
 
     """
-    if coord in ["forecast_reference_time"]:
+    if coord in ["forecast_reference_time", "model"]:
```

Comment:
I think our original implementation here, and this fix are both too simple. I think we need to check if the coordinate being collapsed shares the dimension of the 'forecast_reference_time', in which case, the forecast periods etc. that are being modified within this if statement need to be tackled. Will this cover the case for which you made your original modification?

`if cube.coord_dims(coord) == cube.coord_dims('forecast_reference_time'):`

Does this seem reasonable to you?

Reply:
I think this will work. I just need to check what happens if we blend two models with the same forecast_reference_time (in which case I don't think we need to make any changes but better just check it still works).

Reply:
Unfortunately on making the change suggested quite a few of the unit tests fail.